keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -121,35 +125,36 @@ open class SqlResourceRepository(
   }
 
   // todo: this is not retryable due to overall repository structure: https://github.com/spinnaker/keel/issues/740
-  override fun store(resource: Resource<*>) {
-    val uid = jooq.select(RESOURCE.UID)
+  override fun store(resource: Resource<*>): Resource<*> {
+    val uid = randomUID().toString()
+    val version = jooq
+      .select(
+        coalesce(
+          max(RESOURCE.VERSION).plus(1),
+          value(1)
+        )
+      )
       .from(RESOURCE)
       .where(RESOURCE.ID.eq(resource.id))
-      .fetchOne(RESOURCE.UID)
-      ?: randomUID().toString()
+      .fetchOneInto(Int::class.java)
 
-    val updatePairs = mapOf(
-      RESOURCE.KIND to resource.kind,
-      RESOURCE.ID to resource.id,
-      RESOURCE.SPEC to objectMapper.writeValueAsString(resource.spec),
-      RESOURCE.APPLICATION to resource.application
-    )
-    val insertPairs = updatePairs + (RESOURCE.UID to uid)
-    jooq.insertInto(
-      RESOURCE,
-      *insertPairs.keys.toTypedArray()
-    )
-      .values(*insertPairs.values.toTypedArray())
-      .onDuplicateKeyUpdate()
-      .set(updatePairs)
+    jooq.insertInto(RESOURCE)
+      .set(RESOURCE.UID, uid)
+      .set(RESOURCE.KIND, resource.kind.toString())
+      .set(RESOURCE.ID, resource.id)
+      .set(RESOURCE.VERSION, version)
+      .set(RESOURCE.SPEC, objectMapper.writeValueAsString(resource.spec))
+      .set(RESOURCE.APPLICATION, resource.application)
       .execute()
+
     jooq.insertInto(RESOURCE_LAST_CHECKED)
       .set(RESOURCE_LAST_CHECKED.RESOURCE_UID, uid)
       .set(RESOURCE_LAST_CHECKED.AT, EPOCH.plusSeconds(1))
-      .onDuplicateKeyUpdate()
-      .set(RESOURCE_LAST_CHECKED.AT, EPOCH.plusSeconds(1))
-      .set(RESOURCE_LAST_CHECKED.IGNORE, false)
       .execute()
```

Comment:
Reminder to self: I need to verify that we do _not_ try to check old versions of resources. So long as we read from the `resource_with_metadata` view that won't be the case but having a test in place would be a good idea.

Reply:
Okay I added a test the checks this